contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,217 @@
+package voldemort.store.readonly.mr.azkaban;
+
+import java.util.Map;
+import java.util.List;
+import java.net.URI;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.mapred.JobConf;
+
+import gobblin.runtime.api.JobExecutionResult;
+import gobblin.runtime.embedded.EmbeddedGobblin;
+import gobblin.runtime.embedded.EmbeddedGobblinDistcp;
+
+import voldemort.utils.Props;
+import voldemort.utils.logging.PrefixedLogger;
+
+import azkaban.jobExecutor.AbstractJob;
+
+
+public class GobblinDistcpJob extends AbstractJob {
+    private final static String ATTR_PREFIX = "distcpConf.";
+    private String source;
+    private final String destination;
+    private final Props props;
+
+    GobblinDistcpJob(String id, String sourceHdfsCluster, String destinationVoldemortCluster, Props props) {
+        super(id, PrefixedLogger.getLogger(GobblinDistcpJob.class.getName(), destinationVoldemortCluster));
+        this.source = sourceHdfsCluster;
+        this.destination = destinationVoldemortCluster;
+        this.props = props;
+    }
+
+    public void run() throws Exception {
+        info("###############  Distcp  ###############");
+        try {
+            String cdnURL = pickCDN();
+            if (!cdnURL.isEmpty()) {
+                info("Using CDN: " + cdnURL);
+                String username = props.getString("env.USER", "unknownUser");
+                String pathPrefix = props.getString(VoldemortBuildAndPushJob.PUSH_CDN_PREFIX).trim();
+                assert pathPrefix.startsWith("/");
+                pathPrefix = removeTrailingSlash(pathPrefix);
+                // Replace original cluster with CDN, e.g. hdfs://original:9000/a/b/c => webhdfs://cdn:50070/prefix/user/a/b/c
+                String cdnDir =  cdnURL + pathPrefix + "/" + username + extractPathFromUrl(source);
```

Comment:
Shouldn't we use the storename here?
If we are planning to use 'username' as the file path, there will be a potential risk of resource leaking.
Let us say, a team has 10 members, and anyone of them could run the job in Azkaban, but there will be only one user to run the job after the testing:
1. When all the members are running the job at the same time, they could occupy 10 times of spaces in dropbox, which is not good from my understanding;
2. There is a risk that the user folder won't clean up forever if that user stopped pushing data after the job failure;

Since Voldemort doesn't support parallel push, the copied data in dropbox could be corrupted because of interleaving override, and I think it is acceptable to have the corrupted data in dropbox for this scenario.

But there is a concern with the above proposal:
which user should we use to create the data folder for the given store? If we are going to use the actual job owner, other users may not be able to delete the data file next time.
Since this cdn feature is for optimization, I am thinking to use a global user instead of individual user for all the folders, what do you think?